src/java.base/share/classes/sun/security/util/KnownOIDs.java

```diff
@@ -505,14 +477,23 @@ private static void register(KnownOIDs o) {
         this.oid = oid;
         this.stdName = name(); // defaults to enum name
         this.aliases = new String[0];
+        this.registerNames = true;
     }
 
     KnownOIDs(String oid, String stdName, String... aliases) {
         this.oid = oid;
         this.stdName = stdName;
+        this.registerNames = true;
         this.aliases = aliases;
     }
 
+    KnownOIDs(String oid, String stdName, boolean registerNames) {
+        this.oid = oid;
+        this.stdName = stdName;
+        this.registerNames = registerNames;
+        this.aliases = new String[0];
```

Comment:
Unless this String[] needs to be unique, it would be sufficient to use the same empty array everywhere.

Reply:
you are right, but accessing static field from enum constructor is not allowed

Reply:
You could define the static field on another class, like SecurityConstants, and then use it in the enum constructor.

Reply:
Creating a new empty string array every time is the original behavior, which has nothing to do with the purpose of this PR. Should we follow the principle of minimal changes? If we want to change it, we should submit another PR

Reply:
Fine either way, its a fuzzy cutoff between an incidental improvement (also perf related) with the same reviewers or a new PR with a new set of emails and reviewers.